core/src/main/scala/org/apache/spark/deploy/client/AppClient.scala

```diff
@@ -77,6 +77,16 @@ private[spark] class AppClient(
     private val registrationRetryThread =
       ThreadUtils.newDaemonSingleThreadScheduledExecutor("appclient-registration-retry-thread")
 
+    private val receiveAndReplyMaxPoolSize = conf.getInt(
```

Comment:
I think a configurable max is not necessary and a small thread pool size would be enough. There are only two requests using it: `RequestExecutors` and `KillExecutors`. These messages are used in dynamic allocation. The request rate is very slow.